ui/docs/reference/core/worker-manager/google.md

```diff
@@ -0,0 +1,42 @@
+order: 10
+---
+import SchemaTable from 'taskcluster-ui/components/SchemaTable'
+
+# Google Provider Type
+
+Google-based providers create workers dynamically in GCE, using a single GCP project.
+Best practice is to use a dedicated project for each provider that is not shared with other uses.
+This provides additional security both for the workers and for any other GCP resources that the workers might otherwise accidentally gain access to.
+
+## Provider Configuration
+
+A google-based provider is be configured in `providers` with the following structure:
+
+```json
+{
+  "myProvider": {
+    "providerType": "google",
+    "project": "<gcp project identifier>",
+    "instancePermissions": "<instance permissions>",
+    "creds": "<google credentials>",
+    "credsFile": "<filename containing google credentials>"
+  },
+  ...
+}
+```
+
+The `project` configuration names the project in which instances will be created.
+It is the project *ID* and not the human-readable project name.
+
+The `instancePermissions` configuration defines the permissions granted to the GCP role assumed by the workers.
+TODO: what does this look like? does everyone need it?
```

Comment:
I can help document this better next week but it is an array of strings where each string is a gcp cloud iam permission string. Most users will probably want the permissions to write to stackdriver at least. At this time I can't think of anything else it would want.

Reply:
I summarized that.